### PR TITLE
Add Ior Type

### LINKED ISSUE #28: Create Ior Type
based on https://github.com/typelevel/cats/blob/master/core/src/main/scala/cats/data/Ior.scala

### PR DESCRIPTION
**Pull request by ffgiraldez**
Resolve #28

### CODE REVIEW

File: katz/src/main/kotlin/katz/Ior.kt [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+ * an `A` and a `B`. It is right-biased so methods such as `map` and `flatMap` operate on the
+ * `B` value. Some methods, like `flatMap`, handle the presence of two [Ior.Both] values using a
+ * `[Semigroup]<`A`>, while other methods, like [toEither], ignore the `A` value in a [Ior.Both Both].
+ *
+ * [Ior]<`A`,`B`> is isomorphic to [Either]<[Either]<`A`,`B`>, [Pair]<`A`,`B`>>, but provides methods biased toward `B`
+ * values, regardless of whether the `B` values appear in a [Ior.Right] or a [Ior.Both].
+ * The isomorphic Either form can be accessed via the [unwrap] method.
+ */
+
+sealed class Ior<out A, out B> {
```
Comment (JMPergar) [inline katz/src/main/kotlin/katz/Ior.kt]: HK is merged, add instances ;)

Comment (raulraja) [inline katz/src/main/kotlin/katz/Ior.kt]: We don't have yet an HK2,3... to satisfy type constructors with arity > 1

Comment (JMPergar) [inline katz/src/main/kotlin/katz/Ior.kt]: ouch!

**Comment by raulraja** [comment]
@ffgiraldez needs rebasing before this can be merged, great job!

**Pull request merged**